OloEngine/tests/Rendering/WaterRenderingTest.cpp

```
@@ -0,0 +1,239 @@
+
+// =============================================================================
+// WaterUBO — Alignment & Size
+// =============================================================================
+
+TEST(WaterRendering, WaterUBOAlignment)
+{
+    EXPECT_EQ(sizeof(UBOStructures::WaterUBO) % 16, 0u)
+        << "WaterUBO must be 16-byte aligned for std140";
+}
+
+TEST(WaterRendering, WaterUBOSizeStable)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)